activesupport/lib/active_support/core_ext/enumerable.rb

```diff
@@ -119,7 +134,7 @@ class Array
     alias :orig_sum :sum
 
     def sum(init = nil, &block) #:nodoc:
-      if init.is_a?(Numeric) || first.is_a?(Numeric)
+      if init || first.is_a?(Numeric)
```

Comment:
Actually my mistake, its not interchangeable for all objects. probably Enumerable isn't too, let me see.